replication/binlogsyncer.go

```diff
@@ -163,16 +171,42 @@ func (b *BinlogSyncer) registerSlave() error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+
 	if len(b.cfg.Charset) != 0 {
 		b.c.SetCharset(b.cfg.Charset)
 	}
 
+	//set read timeout
+	if b.cfg.ReadTimeout > 0 {
+		b.c.SetReadDeadline(time.Now().Add(time.Second * b.cfg.ReadTimeout))
+	}
+
 	if b.cfg.RecvBufferSize > 0 {
 		if tcp, ok := b.c.Conn.Conn.(*net.TCPConn); ok {
 			tcp.SetReadBuffer(b.cfg.RecvBufferSize)
 		}
 	}
 
+	// kill last connection id
+	if b.lastConnectionID > 0 {
+		cmd := fmt.Sprintf("KILL %d", b.lastConnectionID)
+		if _, err := b.c.Execute(cmd); err != nil {
+			log.Errorf("b.c.Execute(%s) error(%v)", cmd, err)
+			var tmpStr string
+			var code int
+			// golang scanf doesn't support %*,so I used a temporary variable
+			fmt.Sscanf(err.Error(), "%s%d", &tmpStr, &code)
```

Comment:
can we ensure the message style is string + ID here? If not, I worry the scan may panic here. 
If we can't guarantee this, we may still check error message.